lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -167,7 +171,11 @@ def get(self, request, *args, **kwargs):  # pylint: disable=too-many-statements
         monitoring_utils.set_custom_attribute('user_id', request.user.id)
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
 
-        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        try:
+            course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        except CourseAccessRedirect as access_redirect:
+            serializer = OutlineTabCourseAccessRedirectSerializer(access_redirect)
```

Comment:
I feel this addition needs documentation in code. 
See the comment in the beginning of this function. Could you please explain in there as well that we would return this redirect response in this case? Please spell out the case clearly so other developers won't be confused.

Reply:
I'm going to argue that this is the wrong way to fix this problem. See https://github.com/openedx/frontend-app-learning/pull/1111#pullrequestreview-1424707928